// opmoGenerator/lib/motifList.js
module.exports = {
  renda : {
    //renda-'c6-10
    pitch : {
      func : 'gen-repeat',
      args: ['this.args[1]', 'this.args[0]']
    },
    length : {
      func: `stretch`,
      args: ['this.timeSig', 'pitch', 'this.args[2]']
    },
    articulation:{
      return : '(mclist (span pitch \'(stacc)))'
    },
    velocity: {
      return: '(gen-dynamic (list (length (flatten pitch))) \'(ppp) \'fff)'
    }
  },
  chord : {
    pitch : {
      // return : 'chord1'
      returnEvaled : 'genChord',
      args: ["this.args[0]", "this.args[1]"]
    },
    length : {
      // returnEvaled : 'this.timeSig',
      // listed : true
      func: `length-divide`,
      args: ['1','5', '(rnd-number 1 3 7)']
    }
  },
  ascend : {
    pitch : {
      func: 'ascend',
      args: ['this.args[0]', 'this.args[1]', 'this.args[2]']
    },
    length : {
      func: `stretch`,
      args: ['this.timeSig', 'pitch']
    },
    velocity: {
      return: '(gen-dynamic (list (length (flatten pitch))) \'(ppp) \'fff)'
    },
    articulation: {
      return : '(mclist (span pitch \'(leg)))'
      // return : '\'(pizz)'
    }
  },
  random : {
    pitch: {
      func: 'integer-to-pitch',
      args: ['(rnd-row)']
    },
    length: {
      func: `length-weight`,
      args: ['(span pitch \'(s))']
    },
    // velocity: {
    //   return : '\'(f)'
    // }
  }
};
