// opmoGenerator/lib/Motif.js
var motifList = require('./motifList');

class Motif {
  constructor(timeSig, funcName, args){
    this.timeSig = timeSig;
    this.funcName = funcName;
    this.args = args;
  }
  generate(){
    var ret = "";
    var motifSnippet = motifList[this.funcName];

    var pitchSnippet = this.handleSnippet(motifSnippet.pitch);
    var lengthSnippet = this.handleSnippet(motifSnippet.length);
    var pitch = this.genPitch(pitchSnippet);
    var length = this.genLength(lengthSnippet);
    ret += pitch;
    ret += length;
    return ret;
  }

  handleSnippet(snippet){
    var ret = "";

    //Direct return
    if(snippet.hasOwnProperty('returnEvaled')){
      if(snippet.listed){
        ret += "\'(";
      }
      ret += eval(snippet.returnEvaled);
      if(snippet.listed){
        ret += ")";
      }

    }

    //Callbacks predefined function
    else if (snippet.hasOwnProperty('func')){
      ret += `(${snippet.func} `;

      var args = [];
      snippet.args.forEach((arg)=>{
        if(arg.match('this')){
          args.push(eval(arg));
        }
        else {
          args.push(arg);
        }
      });
      args.forEach((arg)=>{
        ret += `${arg} `;
      });
      ret += `)`;
    }

    return ret;
  }

  genPitch(pitchSnippet){
    return `(setf pitch ${pitchSnippet})\n`;
  }

  genLength(lengthSnippet){
    return `(setf length ${lengthSnippet})\n`;
  }

}


module.exports = Motif;
